src/Build/Construction/ProjectSdkElement.cs

```diff
@@ -0,0 +1,86 @@
+﻿using Microsoft.Build.Internal;
+using Microsoft.Build.Shared;
+
+namespace Microsoft.Build.Construction
+{
+    /// <summary>
+    ///     ProjectSdkElement represents the Sdk element within the MSBuild project.
+    /// </summary>
+    public class ProjectSdkElement : ProjectElementContainer
+    {
+        /// <summary>
+        ///     Initialize a parented ProjectSdkElement
+        /// </summary>
+        internal ProjectSdkElement(XmlElementWithLocation xmlElement, ProjectRootElement parent,
+            ProjectRootElement containingProject)
+            : base(xmlElement, parent, containingProject)
+        {
+            ErrorUtilities.VerifyThrowArgumentNull(parent, "parent");
+        }
+
+        /// <summary>
+        ///     Initialize an non-parented ProjectSdkElement
+        /// </summary>
+        private ProjectSdkElement(XmlElementWithLocation xmlElement, ProjectRootElement containingProject)
+            : base(xmlElement, null, containingProject)
+        { }
+
+        /// <summary>
+        /// Gets or sets the name of the SDK.
+        /// </summary>
+        public string Name
+        {
+            get { return ProjectXmlUtilities.GetAttributeValue(XmlElement, XMakeAttributes.sdkName); }
+            set
+            {
+                ErrorUtilities.VerifyThrowArgumentLength(value, XMakeAttributes.sdkName);
+                ProjectXmlUtilities.SetOrRemoveAttribute(XmlElement, XMakeAttributes.sdkName, value);
+                MarkDirty($"Set SDK Version to {value}", XMakeAttributes.sdkName);
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the version of the SDK.
+        /// </summary>
+        public string Version
+        {
+            get { return ProjectXmlUtilities.GetAttributeValue(XmlElement, XMakeAttributes.sdkVersion); }
+            set
+            {
+                ProjectXmlUtilities.SetOrRemoveAttribute(XmlElement, XMakeAttributes.sdkVersion, value);
+                MarkDirty($"Set SDK Version to {value}", XMakeAttributes.sdkVersion);
+            }
+        }
+
+        /// <inheritdoc />
+        internal override void VerifyThrowInvalidOperationAcceptableLocation(ProjectElementContainer parent,
+            ProjectElement previousSibling, ProjectElement nextSibling)
+        {
+            ErrorUtilities.VerifyThrowInvalidOperation(parent is ProjectRootElement, "OM_CannotAcceptParent");
```

Comment:
The question of whether to throw errors when encountering one of these in an imported file arises again :(